features/kafka/producer/src/main/java/org/opennms/features/kafka/producer/collection/CollectionSetMapper.java

```diff
@@ -0,0 +1,194 @@
+                            .newBuilder();
+                    attributeBuilder.setGroup(lastGroupName);
+                    attributeBuilder.setName(attribute.getName());
+                    attributeBuilder.setValue(attribute.getNumericValue().longValue());
+                    attributeBuilder.setType((attribute.getType() == AttributeType.GAUGE) ? Type.GAUGE : Type.COUNTER);
+                    collectionSetResourceBuilder.addNumeric(attributeBuilder);
+                }
+
+            }
+
+            @Override
+            public void completeAttribute(CollectionAttribute attribute) {
+                // we already set it in visitAttribute
+
+            }
+
+            @Override
+            public void completeGroup(AttributeGroup group) {
+                // group here is part of Numeric, it's already set in numeric
+
+            }
+
+            @Override
+            public void completeResource(CollectionResource resource) {
+                builder.addResource(collectionSetResourceBuilder);
+            }
+
+            @Override
+            public void completeCollectionSet(CollectionSet set) {
+                builder.setTimestamp(collectionSet.getCollectionTimestamp().getTime());
+            }
+
+        });
+
+        return builder.build();
+    }
+
+    private String getNodeCriteriaFromResource(CollectionResource resource) {
+
+        String[] parentResourcePathElements = null;
+        if ( resource != null && resource.getParent() != null ) {
+          parentResourcePathElements = resource.getParent().elements();
+        } else {
+            return null;
+        }
+        String nodeCriteria = null;
+        if (ResourceTypeUtils.FOREIGN_SOURCE_DIRECTORY.equals(parentResourcePathElements[0])) {
+            final String fs = parentResourcePathElements[1];
+            final String fid = parentResourcePathElements[2];
+            nodeCriteria = fs + ":" + fid;
+        } else {
+            final String nodeId = parentResourcePathElements[0];
+            nodeCriteria = nodeId;
+        }
+        return nodeCriteria;
+    }
+
+    
+    public CollectionSetProtos.NodeLevelResource.Builder buildNodeLevelResourceForProto(String nodeCriteria) {
+        CollectionSetProtos.NodeLevelResource.Builder nodeResourceBuilder = CollectionSetProtos.NodeLevelResource
+                .newBuilder();
+        transactionOperations.execute((TransactionCallback<Void>) status -> {
+            try {
+                OnmsNode node = nodeDao.get(nodeCriteria);
+                nodeResourceBuilder.setNodeId(node.getId());
+                nodeResourceBuilder.setNodeLabel(node.getLabel());
+                nodeResourceBuilder.setForeignId(node.getForeignId());
+                nodeResourceBuilder.setForeignSource(node.getForeignSource());
+                nodeResourceBuilder.setLocation(node.getLocation().getLocationName());
+            } catch (Exception e) {
+               //TODO: Deal with response time resources
```

Comment:
We should ensure that response time resource are properly handled before merging this.